src/Microsoft.Diagnostics.EventFlow.Core/Implementations/HealthReporters/CsvHealthReporter.cs

```diff
@@ -418,6 +419,17 @@ private void Initialize(CsvHealthReporterConfiguration configuration, INewReport
 
             this.innerReportWriter = this.ReportText;
 
+            if (configuration.AssumeSharedLogFolder)
+            {
+                var random = new Random(); 
+                const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+                this.logFileNameRandomizer = new string(Enumerable.Range(1, 12).Select(i => chars[random.Next(chars.Length)]).ToArray());
```

Comment:
Did not know about this API, thanks.

However in our case we want to control what goes into the name of the file so that we can find "our" old log files and clean them up, without touching log files that belong to other instances of the service/process/EventFlow writing into the same folder. So Path.GetRandomFileName() is not going to help us here.

Reply:
I don't think knowing what goes to the randomizer helps us identify our files. In theory, the user can always stick something similar to the place and the cleaner will clean it.
I am a fan of making use existing API. That said, **I don't see any hard break** for the current implementation, and you have unit tests covering the implementation as well. I think it is fine we keep it as is unless something unseen turns out broken.